src/redisearch_rs/result_processor/src/test_utils.rs

```diff
@@ -95,16 +135,29 @@ impl Chain {
         P: ResultProcessor + 'static,
     {
         let mut result_processor = ResultProcessorWrapper::new(result_processor);
+        let query_processing_context_ptr: *mut ffi::QueryProcessingCtx =
+            &*self.query_processing_context as *const _ as *mut _;
```

Comment:
lets use as few `as` pointer casts as possible
```suggestion
        let query_processing_context_ptr: *mut ffi::QueryProcessingCtx =
        self.query_processing_context.as_ptr().cast_mut();
```